src/react/databrowser/objects/__tests__/Metadata.test.jsx

```diff
@@ -0,0 +1,102 @@
+        });
+
+        // select a key
+        optionsComponents.at(0).simulate('click');
+
+        // check dropdown menu disappear
+        expect(component.find('Option')).toHaveLength(0);
+
+        // add a value to the key
+        component.find('input#metadata-input-value').simulate('change', { target: { value: 'value1' } });
+
+        // click on add button
+        component.find(AddButton).simulate('click');
+
+        // check new key/value added
+        expect(component.find('SelectBox')).toHaveLength(2);
+
+        // check if function when pressing save button is triggered
+        expect(putObjectMetadataMock).toHaveBeenCalledTimes(0);
+        component.find('Button#metadata-button-save').simulate('click');
+        expect(putObjectMetadataMock).toHaveBeenCalledTimes(1);
+    });
+
+    it('remove button and add button should be disabled', () => {
+        const { component } = reduxMount(<Metadata objectMetadata={OBJECT_METADATA}/>);
+
+        expect(component.find(AddButton).children().prop('disabled')).toBe(true);
+        expect(component.find(SubButton).children().prop('disabled')).toBe(true);
+    });
+
+    it('should render SelectBox with key/value pass in props', () => {
+        const { component } = reduxMount(<Metadata
+            objectMetadata={{ ...OBJECT_METADATA, metadata: [{ key: 'cache-control', value: 'value1' }] }}/>);
+
+        expect(component.find('input[name="mdKeyType"]').prop('value')).toBe('cache-control');
+        expect(component.find('input#metadata-input-value').prop('value')).toBe('value1');
+        expect(component.find('SelectBox')).toHaveLength(1);
+    });
+
+    it('should delete key/value if remove button is pressed', () => {
+        const { component } = reduxMount(<Metadata objectMetadata={{
+            ...OBJECT_METADATA,
+            metadata: [{ key: 'cache-control', value: 'value1' }, { key: 'content-encoding', value: 'value2' }],
+        }}/>);
+
+        // check if everything render well
+        expect(component.find('input[name="mdKeyType"]').at(0).prop('value')).toBe('cache-control');
+        expect(component.find('input#metadata-input-value').at(0).prop('value')).toBe('value1');
+        expect(component.find('input[name="mdKeyType"]').at(1).prop('value')).toBe('content-encoding');
+        expect(component.find('input#metadata-input-value').at(1).prop('value')).toBe('value2');
+        expect(component.find('SelectBox')).toHaveLength(2);
+
+        // check if first AddButton is hidden and second one is visible
+        expect(component.find(AddButton).children().at(0).prop('isVisible')).toBe(false);
+        expect(component.find(AddButton).children().at(1).prop('isVisible')).toBe(true);
+
+        // check if there is only one key/value left after triggered remove button
+        component.find(SubButton).children().at(0).simulate('click');
+        expect(component.find('input[name="mdKeyType"]').prop('value')).toBe('content-encoding');
```

Comment:
We can first check that only one "row" is returned before checking its value.

Reply:
I already check it if you look two lines below:
expect(component.find('SelectBox')).toHaveLength(1);
but maybe it was not clearer so now I check before checking values and I'm looking for Item component and not SelectBox Component